Server/ReasnAPI/ReasnAPI/Services/ParameterService.cs

```diff
@@ -71,6 +71,25 @@ public void DeleteParameter(int parameterId)
         context.Parameters.Remove(parameter);
         context.SaveChanges();
     }
+
+    public void ForceDeleteParameter(int parameterId)
+    {
+        var parameter = context.Parameters.FirstOrDefault(r => r.Id == parameterId);
+        if (parameter is null)
+        {
+            throw new NotFoundException("Parameter not found");
+        }
+
+        var eventsWithParameters = context.Events.Include(e => e.Parameters).ToList();
+        foreach (var eventWithParameters in eventsWithParameters)
+        {
+            eventWithParameters.Parameters.Remove(parameter);
+        }
```

Comment:
Isn't it an overkill to get ALL events and check one by one, maybe some have this param, maybe not. If we want to delete param and it's only on 1 event, we shouldn't select 1000000 events from DB

Reply:
changed